src/renderer/base/thread.cpp

```diff
@@ -213,12 +213,6 @@ DWORD WINAPI RenderThread::_ThreadProc()
         LOG_IF_FAILED(_pRenderer->PaintFrame());
 
         SetEvent(_hPaintCompletedEvent);
-
-        // extra check before we sleep since it's a "long" activity, relatively speaking.
-        if (_fKeepRunning)
-        {
-            Sleep(s_FrameLimitMilliseconds);
-        }

```

Comment:
I thought about this today and I will solve the issue by reducing the maximum time DxEngine and AtlasEngine may wait in any blocking event from 1000ms down to 100ms (because 2 frames at 23.97 Hz are roughly 85ms).
Combined with a sleep duration of just 8ms I'm positive we won't be running into issues with callers of `WaitForPaintCompletionAndDisable`. So far only the WDDM engine is specifying a non-infinite timeout of 1000ms.